src/Plugin/os2web/DataLookup/DataLookupBase.php

```diff
@@ -22,42 +23,56 @@ abstract class DataLookupBase extends PluginBase implements DataLookupInterface
    *
    * @var bool
    */
-  protected $isReady = TRUE;
+  protected bool $isReady = TRUE;
+
+  /**
+   * Audit logger.
+   *
+   * @var \Drupal\os2web_audit\Service\Logger
+   */
+  protected Logger $auditLogger;
 
   /**
    * {@inheritdoc}
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    Logger $auditLogger,
+  ) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->auditLogger = $auditLogger;
```

Comment:
Consider using https://stitcher.io/blog/constructor-promotion-in-php-8 here:

```suggestion
    protected readonly Logger $auditLogger,
  ) {
    parent::__construct($configuration, $plugin_id, $plugin_definition);
```

Reply:
I did not do it that way to match the standard Drupal way that the rest of the code uses